va/va.go

```diff
@@ -463,14 +464,10 @@ func (va *ValidationAuthorityImpl) performRemoteValidation(
 
 	responses := make(chan *response, remoteVACount)
 	for _, i := range rand.Perm(remoteVACount) {
-		go func(rva RemoteVA, out chan<- *response) {
+		go func(rva RemoteVA) {
 			res, err := rva.PerformValidation(ctx, req)
-			out <- &response{
-				addr:   rva.Address,
-				result: res,
-				err:    err,
-			}
-		}(va.remoteVAs[i], responses)
+			responses <- &response{rva.Address, res, err}
+		}(va.remoteVAs[i])
```

Comment:
Just noting that removing the channel as an argument to this anonymous function is not a necessary change.

I added the channel as an argument here in https://github.com/letsencrypt/boulder/pull/7522 upon Jacob's reminder that "explicit is better than implicit" -- it's better for the anonymous function to be explicit about what objects it is using than to implicitly close over everything.

It's a fully arbitrary line -- obviously I added the channel as an argument but didn't bother to add `ctx` as an argument too. I think there are perfectly good arguments to be made that this anonymous function should take no arguments, just the rva (effectively the loop variable), or take _everything_ as arguments. Just noting that this is a personal preference change for which we don't have an established team-wide best practice.

Reply:
Also worth noting I recently added a `subCtx` here (which is captured in the same way as `ctx` is) and considered adding it to the explicit argument list but wound up not doing so, since I thought folks might not like it stylistically. So there's some local pattern matching possible for "does this anonymous function need to receive all things as parameters? or is taking some of them as captures okay?"

Reply:
Oh and I remembered the specific reason I backed out that change! I was going to pass `subCtx` as a parameter, but call that parameter `ctx` locally. So inside the function we would have `ctx` shadowed and have no chance of using the original `ctx` by accident. But I decided that was too clever by half and backed it out.

Reply:
Ah, interesting! Personally, I prefer closing over everything. Initially, I wasn’t planning to backport this change and was instead going to update the MPIC code. However, I noticed we’re doing the same thing in this function’s offshoot in va/caa.go, so I figured this was a consistency win:

https://github.com/letsencrypt/boulder/blob/a8cdaf89897df2814252f2cf7e793c4ef29e8116/va/caa.go#L239-L273